pkg/storage/backend-storage/backend-storage.go

```diff
@@ -165,48 +326,33 @@ func (bs *BackendStorage) getAccessMode(storageClass string, mode v1.PersistentV
 	return accessMode
 }
 
-func updateVolumeStatus(vmi *corev1.VirtualMachineInstance, accessMode v1.PersistentVolumeAccessMode) {
+func (bs *BackendStorage) UpdateVolumeStatus(vmi *corev1.VirtualMachineInstance, pvc *v1.PersistentVolumeClaim) {
```

Comment:
This function corrupts. See:

```
diff --git a/pkg/virt-controller/watch/vmi/vmi_test.go b/pkg/virt-controller/watch/vmi/vmi_test.go
index c491d6c57d..68d0ff06e9 100644
--- a/pkg/virt-controller/watch/vmi/vmi_test.go
+++ b/pkg/virt-controller/watch/vmi/vmi_test.go
@@ -40,6 +40,7 @@ import (
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	k8sruntime "k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/kubernetes/fake"
@@ -303,6 +304,32 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 		Expect(controller.dataVolumeIndexer.Add(dv)).To(Succeed())
 	}
 
+	deepCopyList := func(objects []interface{}) []interface{} {
+		for i := range objects {
+			objects[i] = objects[i].(runtime.Object).DeepCopyObject()
+		}
+		return objects
+	}
+
+	sanityExecute := func() {
+		stores := []cache.Store{
+			controller.vmiIndexer, controller.cdiStore, controller.cdiConfigStore, controller.dataVolumeIndexer,
+			controller.podIndexer, controller.pvcIndexer, controller.vmStore,
+		}
+
+		listOfObjects := [][]interface{}{}
+
+		for _, store := range stores {
+			listOfObjects = append(listOfObjects, deepCopyList(store.List()))
+		}
+
+		controller.Execute()
+
+		for i, objects := range listOfObjects {
+			ExpectWithOffset(1, stores[i].List()).To(ConsistOf(objects...))
+		}
+	}
+
 	Context("On valid VirtualMachineInstance given with DataVolume source", func() {
 
 		dvVolumeSource := virtv1.VolumeSource{
@@ -325,7 +352,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addDataVolumePVC(dvPVC)
 			addDataVolume(dataVolume)
 
-			controller.Execute()
+			sanityExecute()
 			testutils.ExpectEvent(recorder, kvcontroller.SuccessfulCreatePodReason)
 			expectMatchingPodCreation(vmi)
 			expectVMIDataVolumeReadyCondition(vmi.Namespace, vmi.Name, k8sv1.ConditionTrue)
@@ -360,7 +387,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				Equal("/bin/bash -c echo bound PVCs"),
 			)
 
-			controller.Execute()
+			sanityExecute()
 			testutils.ExpectEvent(recorder, kvcontroller.SuccessfulCreatePodReason)
 
 			expectMatchingPodCreation(vmi, IsPodWithoutVmPayload)
@@ -396,7 +423,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(pod)
 			addDataVolume(dataVolume)
-			controller.Execute()
+			sanityExecute()
 			expectVMIBeInPhase(vmi.Namespace, vmi.Name, virtv1.Pending)
 			expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name, ContainElement(virtv1.VirtualMachineInstanceCondition{
 				Type:   virtv1.VirtualMachineInstanceProvisioning,
@@ -425,7 +452,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addActivePods(vmi, pod.UID, "")
 			addDataVolume(dataVolume)
 
-			controller.Execute()
+			sanityExecute()
 			testutils.ExpectEvent(recorder, kvcontroller.SuccessfulDeletePodReason)
 			expectVMIDataVolumeReadyCondition(vmi.Namespace, vmi.Name, k8sv1.ConditionTrue)
 			expectPodDoesNotExist(pod.Namespace, pod.Name)
@@ -485,7 +512,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				addActivePods(vmi, pod.UID, "")
 				addDataVolume(dataVolume)
 
-				controller.Execute()
+				sanityExecute()
 				expectVMIBeInPhase(vmi.Namespace, vmi.Name, expectedPhase)
 			},
 			Entry("fail if pod in failed state", k8sv1.PodFailed, nil, virtv1.Failed),
@@ -514,7 +541,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addDataVolumePVC(dvPVC)
 			addDataVolume(dataVolume)
-			controller.Execute()
+			sanityExecute()
 			expectVMIDataVolumeReadyCondition(vmi.Namespace, vmi.Name, k8sv1.ConditionFalse)
 		})
 
@@ -549,13 +576,14 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				pvc.Spec.AccessModes = []k8sv1.PersistentVolumeAccessMode{k8sv1.ReadWriteMany}
 				addDataVolumePVC(pvc)
 
-				controller.Execute()
+				sanityExecute()
 				expectVirtualMachinePendingState(vmi.Namespace, vmi.Name)
 			},
 				Entry("using explicit Immediate mode", pointer.P(storagev1.VolumeBindingImmediate)),
 				Entry("using nil mode", nil))
 
-			It("should create a corresponding Pod on VMI creation with a storage class in WaitForFirstConsumer mode", func() {
+			// Backend storage
+			PIt("should create a corresponding Pod on VMI creation with a storage class in WaitForFirstConsumer mode", func() {
 				sc := &storagev1.StorageClass{
 					ObjectMeta: metav1.ObjectMeta{
 						Name: "testsc456",
@@ -571,7 +599,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				pvc.Spec.AccessModes = []k8sv1.PersistentVolumeAccessMode{k8sv1.ReadWriteMany}
 				addDataVolumePVC(pvc)
 
-				controller.Execute()
+				sanityExecute()
 				testutils.ExpectEvent(recorder, kvcontroller.SuccessfulCreatePodReason)
 				expectMatchingPodCreation(vmi)
 			})
@@ -601,7 +629,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addDataVolumePVC(dvPVC)
 			addDataVolume(dataVolume)
 
-			controller.Execute()
+			sanityExecute()
 			testutils.ExpectEvent(recorder, kvcontroller.SuccessfulCreatePodReason)
 			expectMatchingPodCreation(vmi)
 			expectVMIDataVolumeReadyCondition(vmi.Namespace, vmi.Name, k8sv1.ConditionTrue)
@@ -623,7 +651,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addDataVolumePVC(dvPVC)
 			addDataVolume(dataVolume)
 
-			controller.Execute()
+			sanityExecute()
 			testutils.ExpectEvent(recorder, kvcontroller.SuccessfulCreatePodReason)
 			IsPodWithoutVmPayload := WithTransform(
 				func(pod *k8sv1.Pod) string {
@@ -666,7 +694,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addActivePods(vmi, pod.UID, "")
 			addDataVolume(dataVolume)
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIBeInPhase(vmi.Namespace, vmi.Name, virtv1.Pending)
 			expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name, ContainElement(virtv1.VirtualMachineInstanceCondition{
 				Type:   virtv1.VirtualMachineInstanceProvisioning,
@@ -697,7 +725,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addActivePods(vmi, pod.UID, "")
 			addDataVolume(dataVolume)
 
-			controller.Execute()
+			sanityExecute()
 			testutils.ExpectEvent(recorder, kvcontroller.SuccessfulDeletePodReason)
 			expectPodDoesNotExist(pod.Namespace, pod.Name)
 			expectVMIDataVolumeReadyCondition(vmi.Namespace, vmi.Name, k8sv1.ConditionTrue)
@@ -720,7 +748,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addDataVolumePVC(dvPVC)
 			addDataVolume(dataVolume)
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIDataVolumeReadyCondition(vmi.Namespace, vmi.Name, k8sv1.ConditionFalse)
 		})
 
@@ -737,7 +765,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addDataVolumePVC(pvc)
 
-			controller.Execute()
+			sanityExecute()
 			testutils.ExpectEvent(recorder, kvcontroller.SuccessfulCreatePodReason)
 			expectMatchingPodCreation(vmi)
 		})
@@ -765,7 +793,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			controller.netAnnotationsGenerator = stubNetworkAnnotationsGenerator{
 				annotations: map[string]string{key1: value1, key2: value2},
 			}
-			controller.Execute()
+			sanityExecute()
 			expectPodAnnotations(pod, HaveKeyWithValue(key1, value1), HaveKeyWithValue(key2, value2))
 		})
 
@@ -780,7 +808,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addActivePods(vmi, pod.UID, "")
 
 			controller.netAnnotationsGenerator = stubNetworkAnnotationsGenerator{annotations: map[string]string{}}
-			controller.Execute()
+			sanityExecute()
 
 			expectPodAnnotations(pod, Not(HaveKeyWithValue(key1, value1)), Not(HaveKeyWithValue(key2, value2)))
 		})
@@ -792,7 +820,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 			addVirtualMachine(vmi)
 
-			controller.Execute()
+			sanityExecute()
 
 			testutils.ExpectEvent(recorder, kvcontroller.SuccessfulCreatePodReason)
 			expectMatchingPodCreation(vmi, WithTransform(
@@ -813,7 +841,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 
 			expectPodAnnotations(pod, HaveKey(descheduler.EvictOnlyAnnotation))
 		})
@@ -859,7 +887,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			_, err := kubeClient.CoreV1().Pods(pod3.Namespace).Create(context.Background(), pod3, metav1.CreateOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
-			controller.Execute()
+			sanityExecute()
 
 			testutils.ExpectEvent(recorder, kvcontroller.SuccessfulDeletePodReason)
 			testutils.ExpectEvent(recorder, kvcontroller.SuccessfulDeletePodReason)
@@ -892,7 +920,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addPod(pod)
 			addActivePods(vmi, pod.UID, "")
 
-			controller.Execute()
+			sanityExecute()
 
 			testutils.ExpectEvent(recorder, kvcontroller.SuccessfulDeletePodReason)
 			expectPodDoesNotExist(pod.Namespace, pod.Name)
@@ -906,7 +934,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			controller.podIndexer.Update(modifiedPod)
 			controller.podExpectations.SetExpectations(key, 0, 0)
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIFailedState(vmi)
 		})
 		DescribeTable("should not delete the corresponding Pod if the vmi is in", func(phase virtv1.VirtualMachineInstancePhase) {
@@ -920,7 +948,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addPod(pod)
 			addActivePods(vmi, pod.UID, "")
 
-			controller.Execute()
+			sanityExecute()
 			expectPodExists(pod.Namespace, pod.Name)
 		},
 			Entry("succeeded state", virtv1.Failed),
@@ -933,7 +961,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 			addVirtualMachine(vmi)
 
-			controller.Execute()
+			sanityExecute()
 			Expect(virtClientset.Actions()).To(HaveLen(1))
 			Expect(virtClientset.Actions()[0].GetVerb()).To(Equal("create"))
 			Expect(virtClientset.Actions()[0].GetResource().Resource).To(Equal("virtualmachineinstances"))
@@ -949,7 +977,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				return true, nil, fmt.Errorf("random error")
 			})
 
-			controller.Execute()
+			sanityExecute()
 
 			testutils.ExpectEvent(recorder, kvcontroller.FailedCreatePodReason)
 			expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name, ContainElement(MatchFields(IgnoreExtras,
@@ -969,7 +997,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				return true, nil, fmt.Errorf("random error")
 			})
 
-			controller.Execute()
+			sanityExecute()
 			Expect(controller.Queue.Len()).To(Equal(0))
 			Expect(mockQueue.GetRateLimitedEnqueueCount()).To(Equal(1))
 
@@ -989,7 +1017,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 			addVirtualMachine(vmi)
 
-			controller.Execute()
+			sanityExecute()
 			Expect(mockQueue.GetRateLimitedEnqueueCount()).To(Equal(0))
 			testutils.ExpectEvent(recorder, kvcontroller.SuccessfulCreatePodReason)
 			expectMatchingPodCreation(vmi)
@@ -1043,7 +1071,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				}
 				addVirtualMachine(vmi)
 
-				controller.Execute()
+				sanityExecute()
 				updatedVMI, err := virtClientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
 				expectConditions(updatedVMI)
@@ -1058,7 +1086,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				// in previous loop we have updated the status
 				controller.vmiExpectations.SetExpectations(key, 0, 0)
 
-				controller.Execute()
+				sanityExecute()
 				Expect(controller.Queue.Len()).To(Equal(0))
 				Expect(mockQueue.GetRateLimitedEnqueueCount()).To(BeZero())
 				// make sure that during next iteration we do not add the same condition again
@@ -1097,7 +1125,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addPod(pod)
 			addActivePods(vmi, pod.UID, "")
 
-			controller.Execute()
+			sanityExecute()
 			expectVMISchedulingState(vmi)
 		},
 			Entry(", not ready and in running state", k8sv1.PodRunning, false),
@@ -1130,7 +1158,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				addVirtualMachine(vmi)
 				addPod(pod)
 
-				controller.Execute()
+				sanityExecute()
 				expectVMIBeInPhase(vmi.Namespace, vmi.Name, virtv1.Scheduling)
 				expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name, ContainElement(MatchFields(IgnoreExtras,
 					Fields{
@@ -1162,7 +1190,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 				ignorePodUpdates()
 
-				controller.Execute()
+				sanityExecute()
 				testutils.IgnoreEvents(recorder)
 				expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name, Not(ContainElement(MatchFields(IgnoreExtras,
 					Fields{
@@ -1182,7 +1210,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 			addVirtualMachine(vmi)
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIFailedState(vmi)
 		})
 		It("should move the vmi to failed state if the vmi is pending, no pod exists yet and gets deleted", func() {
@@ -1191,7 +1219,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 			addVirtualMachine(vmi)
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIFailedState(vmi)
 		})
 
@@ -1208,7 +1236,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				addActivePods(vmi, pod.UID, "")
 			}
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIScheduledState(vmi)
 
 			updatedVMI, err := virtClientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
@@ -1237,7 +1265,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addPod(pod)
 			addActivePods(vmi, pod.UID, "")
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIBeInPhase(vmi.Namespace, vmi.Name, virtv1.Scheduling)
 			expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name, ContainElement(MatchFields(IgnoreExtras,
 				Fields{"Type": Equal(virtv1.VirtualMachineInstanceReady), "Status": Equal(k8sv1.ConditionFalse), "Reason": Equal(virtv1.GuestNotRunningReason)})))
@@ -1296,7 +1324,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addPod(pod)
 			addPod(attachmentPod)
 
-			controller.Execute()
+			sanityExecute()
 
 			switch expectedPhase {
 			case virtv1.Scheduled:
@@ -1357,7 +1385,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 
 			testutils.ExpectEvent(recorder, kvcontroller.FailedDeletePodReason)
 			expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name, ContainElement(MatchFields(IgnoreExtras,
@@ -1390,7 +1418,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 
 			testutils.ExpectEvent(recorder, kvcontroller.FailedGuaranteePodResourcesReason)
 			expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name, ContainElement(MatchFields(IgnoreExtras,
@@ -1411,7 +1439,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 			addVirtualMachine(vmi)
 
-			controller.Execute()
+			sanityExecute()
 
 			testutils.ExpectEvent(recorder, kvcontroller.FailedCreatePodReason)
 			expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name, ContainElement(MatchFields(IgnoreExtras,
@@ -1431,7 +1459,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIScheduledState(vmi)
 		})
 		It("should update the virtual machine QOS class if the pod finally has a QOS class assigned", func() {
@@ -1444,7 +1472,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 
 			updatedVmi, err := virtClientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
@@ -1460,7 +1488,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addPod(pod)
 			addActivePods(vmi, pod.UID, "")
 
-			controller.Execute()
+			sanityExecute()
 
 			pod = newPodForVirtualMachine(vmi, k8sv1.PodRunning)
 
@@ -1471,7 +1499,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			controller.podIndexer.Update(pod)
 			controller.podExpectations.SetExpectations(key, 0, 0)
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIScheduledState(vmi)
 		})
 		It("should update the virtual machine to failed if pod was not ready, triggered by pod delete", func() {
@@ -1484,7 +1512,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addPod(pod)
 			addActivePods(vmi, pod.UID, "")
 
-			controller.Execute()
+			sanityExecute()
 
 			key, err := kvcontroller.KeyFunc(vmi)
 			Expect(err).To(Not(HaveOccurred()))
@@ -1492,7 +1520,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 			controller.podIndexer.Delete(pod)
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIFailedState(vmi)
 		})
 		It("should update the virtual machine to failed if compute container is terminated while still scheduling", func() {
@@ -1505,7 +1533,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIFailedState(vmi)
 		})
 		DescribeTable("should remove the fore ground finalizer if no pod is present and the vmi is in ", func(phase virtv1.VirtualMachineInstancePhase) {
@@ -1519,7 +1547,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 			addVirtualMachine(vmi)
 
-			controller.Execute()
+			sanityExecute()
 
 			updatedVmi, err := virtClientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
@@ -1558,7 +1586,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 			addVirtualMachine(vmi)
 
-			controller.Execute()
+			sanityExecute()
 
 			updatedVmi, err := virtClientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
@@ -1589,7 +1617,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addPod(pod)
 			addActivePods(vmi, pod.UID, "")
 
-			controller.Execute()
+			sanityExecute()
 			Expect(virtClientset.Actions()).To(HaveLen(1))
 			Expect(virtClientset.Actions()[0].GetVerb()).To(Equal("create"))
 			Expect(virtClientset.Actions()[0].GetResource().Resource).To(Equal("virtualmachineinstances"))
@@ -1620,7 +1648,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addActivePods(vmi, pod.UID, "")
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 
 			updatedVmi, err := virtClientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
@@ -1655,7 +1683,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addActivePods(vmi, pod.UID, "")
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 
 			updatedVmi, err := virtClientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
@@ -1674,7 +1702,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addActivePods(vmi, pod.UID, "")
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name,
 				ContainElement(
 					MatchFields(IgnoreExtras, Fields{
@@ -1696,7 +1724,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addActivePods(vmi, pod.UID, "")
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 			updatedVmi, err := virtClientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
 			Expect(updatedVmi.Status.Conditions).To(HaveLen(1))
@@ -1717,7 +1745,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 			updatedVmi, err := virtClientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
 			Expect(updatedVmi.Status.ActivePods).To(HaveLen(1))
@@ -1737,7 +1765,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name, ContainElement(MatchFields(IgnoreExtras,
 				Fields{"Type": Equal(virtv1.VirtualMachineInstanceSynchronized), "Status": Equal(k8sv1.ConditionFalse)})))
 		})
@@ -1748,7 +1776,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 			addVirtualMachine(vmi)
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIBeInPhase(vmi.Namespace, vmi.Name, virtv1.Failed)
 		},
 			Entry("and the vmi is in running state", virtv1.Running),
@@ -1764,7 +1792,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIFailedState(vmi)
 		},
 			Entry("and in succeeded state", k8sv1.PodSucceeded),
@@ -1788,7 +1816,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 			expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name, ContainElement(MatchFields(IgnoreExtras,
 				Fields{
 					"Type":   Equal(virtv1.VirtualMachineInstanceSynchronized),
@@ -1827,7 +1855,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 
 			expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name, ContainElement(MatchFields(IgnoreExtras,
 				Fields{
@@ -1850,7 +1878,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addPod(pod)
 			addActivePods(vmi, pod.UID, "")
 
-			controller.Execute()
+			sanityExecute()
 			updatedVmi, err := virtClientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
 			Expect(updatedVmi.Status.MigrationTransport).To(Equal(virtv1.MigrationTransportUnix))
@@ -1980,7 +2008,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 
 			updatedPod, err := kubeClient.CoreV1().Pods(pod.Namespace).Get(context.Background(), pod.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
@@ -2005,7 +2033,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(pod)
 
-			controller.Execute()
+			sanityExecute()
 
 			updatedPod, err := kubeClient.CoreV1().Pods(pod.Namespace).Get(context.Background(), pod.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
@@ -2043,7 +2071,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				addVirtualMachine(vmi)
 				addPod(pod)
 
-				controller.Execute()
+				sanityExecute()
 				expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name, ContainElement(MatchFields(IgnoreExtras,
 					Fields{
 						"Type":   BeEquivalentTo(virtv1.VirtualMachineInstanceMemoryChange),
@@ -2957,7 +2985,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(virtlauncherPod)
 
-			controller.Execute()
+			sanityExecute()
 			testutils.ExpectEvent(recorder, kvcontroller.SuccessfulCreatePodReason)
 			expectHotplugPod()
 			updatedVmi, err := virtClientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
@@ -3066,7 +3094,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 			addPod(virtlauncherPod)
 			addPod(hpPod)
-			controller.Execute()
+			sanityExecute()
 
 			testutils.ExpectEvent(recorder, kvcontroller.SuccessfulDeletePodReason)
 			updatedVmi, err := virtClientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
@@ -3233,7 +3261,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			It("invtsc feature exists", func() {
 				vmi := getVmiWithInvTsc()
 				addVirtualMachine(vmi)
-				controller.Execute()
+				sanityExecute()
 				expectTopologyHintsDefined(vmi, BeTrue())
 			})
 
@@ -3247,7 +3275,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				When("TSC frequency is exposed", func() {
 					It("topology hints need to be set", func() {
 						addVirtualMachine(vmi)
-						controller.Execute()
+						sanityExecute()
 						expectTopologyHintsDefined(vmi, BeTrue())
 					})
 				})
@@ -3264,7 +3292,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 						unexposeTscFrequency(topology.RequiredForMigration)
 						// Make sure no update occurs
 						addVirtualMachine(vmi)
-						controller.Execute()
+						sanityExecute()
 						expectTopologyHintsDefined(vmi, BeFalse())
 						expectMatchingPodCreation(vmi)
 						testutils.ExpectEvent(recorder, kvcontroller.SuccessfulCreatePodReason)
@@ -3282,7 +3310,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				Expect(topology.GetTscFrequencyRequirement(vmi).Type).To(Equal(topology.RequiredForBoot))
 
 				addVirtualMachine(vmi)
-				controller.Execute()
+				sanityExecute()
 				expectTopologyHintsDefined(vmi, BeTrue())
 			})
 
@@ -3291,7 +3319,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				Expect(topology.GetTscFrequencyRequirement(vmi).Type).To(Equal(topology.RequiredForMigration))
 				addVirtualMachine(vmi)
 
-				controller.Execute()
+				sanityExecute()
 				expectTopologyHintsDefined(vmi, BeTrue())
 			})
 
@@ -3300,7 +3328,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				Expect(topology.GetTscFrequencyRequirement(vmi).Type).To(Equal(topology.NotRequired))
 				addVirtualMachine(vmi)
 
-				controller.Execute()
+				sanityExecute()
 
 				testutils.ExpectEvent(recorder, kvcontroller.SuccessfulCreatePodReason)
 				expectMatchingPodCreation(vmi)
@@ -3318,7 +3346,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 			addVirtualMachine(vmi)
 			addPod(pod)
-			controller.Execute()
+			sanityExecute()
 
 			expectVMIScheduledState(vmi)
 			updatedVmi, err := virtClientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
@@ -3337,7 +3365,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			addVirtualMachine(vmi)
 
 			Expect(controller.vmiIndexer.Delete(vmi)).To(Succeed())
-			controller.Execute()
+			sanityExecute()
 
 			Expect(alc.calls).To(ConsistOf([]string{"Allocate", "Remove"}))
 		})
@@ -3553,7 +3581,7 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				addDataVolume(dataVolume2)
 				addDataVolume(dataVolume3)
 
-				controller.Execute()
+				sanityExecute()
 				testutils.ExpectEvent(recorder, kvcontroller.SuccessfulCreatePodReason)
 				expectMatchingPodCreation(vmi)
 				expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name,

```

Reply:
Nice catch, thank you, should be fixed now

Reply:
Introduced https://github.com/kubevirt/kubevirt/pull/13068 to make this easy to catch (:crossed_fingers: )